localstack/utils/collections.py

```diff
@@ -404,3 +404,12 @@ def is_none_or_empty(obj: Union[Optional[str], Optional[list]]) -> bool:
 def select_from_typed_dict(typed_dict: Type[TypedDict], obj: Dict):
     """Select a subset of attributes from a dictionary based on the keys of a given `TypedDict`"""
     return select_attributes(obj, [*typed_dict.__required_keys__, *typed_dict.__optional_keys__])
+
+
+def sort_dict(structure: Union[Dict, List]) -> Union[Dict, List]:
+    """Sorts all the keys in a dictionary-based structure"""
+    if isinstance(structure, dict):
+        return {k: sort_dict(v) for k, v in sorted(structure.items())}
+    if isinstance(structure, list):
+        return [sort_dict(v) for i, v in sorted(enumerate(structure))]
```

Comment:
Did a quick check with different values for `structure` here - the problem seems to be that `sorted(enumerate(..))` would sort based on the indexes (first item of the `<index, item>` tuple), effectively leaving the list in the same sort order as the original list.

This may not be easy to fix in the general case, because if we remove the `enumerate(..)`, and if the list contains `dict`s, then we may end up with errors like:
```
TypeError: '<' not supported between instances of 'dict' and 'dict'
```

Fortunately, we already have [`canonical_json(..)`](https://github.com/localstack/localstack/blob/eba8880d4a10dac0e286294d7c2d279ad7b117bf/localstack/utils/json.py#L201-L202) util function which - I think - we can use here. It takes an arbitrary structure, and converts it to a canonicalized string representation. Would that help here?

Reply:
Indeed, was trying to reinvent the wheel here ^^. `canonical_json` does exactly what I was trying to achieve 🙏